crates/hir-ty/src/infer.rs

```diff
@@ -412,6 +412,7 @@ pub(crate) struct InferenceContext<'a> {
     /// closures, but currently this is the only field that will change there,
     /// so it doesn't make sense.
     return_ty: Ty,
+    resume_yield_tys: Option<(Ty, Ty)>,
```

Comment:
Comment might be nice about which one the yield and return type is

Reply:
Added doc comment